shopinvader_utm/services/service.py

```diff
@@ -0,0 +1,19 @@
+# Copyright 2021 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo.addons.component.core import AbstractComponent
+
+
+class BaseShopinvaderService(AbstractComponent):
+    _inherit = "base.shopinvader.service"
+
+    def _handle_utm(self, record):
+        if "utm.mixin" in record._inherit_module and hasattr(self.work, "utm"):
+            record.add_utm(self.work.utm)
+
+    def dispatch(self, method_name, *args, params=None):
+        res = super().dispatch(method_name, *args, params=params)
+        if self._expose_model and res.get("data", False):
+            record = self.env[self._expose_model].browse(res["data"]["id"])
```

Comment:
That could break because` res["data"]` couldn't have an key named `id`.
You should do a check first to ensure you have it, then load it via the `_expose_model` like you did.